### ISSUE TITLE
Make Template Name clickable to open details

### ISSUE THREAD

**Issue by annezazu** [WordPress/gutenberg#28243]
## What problem does this address?

When using the Site Editor experiment, to get to the template details, you have to click on a quite small down arrow in order to open the information. [In slack this came up](https://wordpress.slack.com/archives/C015GUFFC00/p1610604957001200) as being fairly confusing as one expected to be able to open template details by clicking on the template name. 

Here's a gif of the current experience:

![temp details](https://user-images.githubusercontent.com/26996883/104757301-20716280-571a-11eb-9c9d-5fa63fd984ef.gif)

## What is your proposed solution?

Allow the template name to be clickable to open up template details. A larger clickable area should also help with accessibility.

**Comment by jameskoster** [comment]
Iirc the a11y issue here is that using the template name as a button label was not adequately descriptive of what clicking the button would actually do.

So the root of the problem seems to be that this component was originally performing two irreconcilable tasks from the a11y point of view:

1. Communicating which template is being edited
2. Acting as a link to view details of said template

We could combine the chevron and template name into a single button if the label was something like "View `template name` details", but that is unnecessarily verbose for task 1 outlined above, and doesn't play nicely when template part names are displayed here too. Hence consensus was found around separating the element in to two parts – label and button. It is a compromise that I would love to find a solution for!

**Comment by oglekler** [comment]
I am agreeing with Anne, I am trying to click on the name and getting nothing. I think for most users it will be confusing and not very convenient. The button is only 24px in width and can be easily missed. Possibly there can be some compromise to make the whole area clickable and at the same time not make the template name itself a button.

**Comment by annezazu** [comment]
Noting that this is still an issue in 5.9. In the Site Editor, it shows as the above. In the Template editor via the Post Editor though, it does allow the full name to be clickable as shown below:


https://user-images.githubusercontent.com/26996883/152718380-9d317933-1f80-4477-9b3d-ef837ff94aa0.mov

**Comment by annezazu** [comment]
This has since evolved immensely leading to the current experience in 6.3 with this title bar: https://github.com/WordPress/gutenberg/issues/50378 Closing this out as a result.
